tests/sensors/test_external_task_sensor.py

```diff
@@ -838,6 +839,172 @@ def test_external_task_group_when_there_is_no_TIs(self):
                 ignore_ti_state=True,
             )
 
+    @pytest.mark.parametrize(
+        "soft_fail, expected_exception, kwargs, expected_message",
+        (
+            (
+                False,
+                AirflowException,
+                {
+                    "external_task_ids": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                f"Some of the external tasks {re.escape(str([TEST_TASK_ID, TEST_TASK_ID_ALTERNATE]))}"
+                f" in DAG {TEST_DAG_ID} failed.",
+            ),
+            (
+                False,
+                AirflowException,
+                {
+                    "external_task_group_id": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                f"The external task_group '{re.escape(str([TEST_TASK_ID, TEST_TASK_ID_ALTERNATE]))}'"
+                f" in DAG '{TEST_DAG_ID}' failed.",
+            ),
+            (
+                False,
+                AirflowException,
+                {"failed_states": [State.FAILED]},
+                f"The external DAG {TEST_DAG_ID} failed.",
+            ),
+            (
+                True,
+                AirflowSkipException,
+                {
+                    "external_task_ids": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                "Skipping due to soft_fail is set to True.",
+            ),
+            (
+                True,
+                AirflowSkipException,
+                {
+                    "external_task_group_id": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
+                    "failed_states": [State.FAILED],
+                },
+                "Skipping due to soft_fail is set to True.",
+            ),
+            (
+                True,
+                AirflowSkipException,
+                {"failed_states": [State.FAILED]},
+                "Skipping due to soft_fail is set to True.",
+            ),
+        ),
+    )
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        "soft_fail, expected_exception",
        (
            (False, AirflowException),
            (True, AirflowSkipException)
        )
    )
    @pytest.mark.parametrize(
        "soft_fail, expected_exception, kwargs, expected_failed_message",
        (
            (
                {
                    "external_task_ids": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
                    "failed_states": [State.FAILED],
                },
                f"Some of the external tasks {re.escape(str([TEST_TASK_ID, TEST_TASK_ID_ALTERNATE]))}"
                f" in DAG {TEST_DAG_ID} failed.",
            ),
            (
                {
                    "external_task_group_id": [TEST_TASK_ID, TEST_TASK_ID_ALTERNATE],
                    "failed_states": [State.FAILED],
                },
                f"The external task_group '{re.escape(str([TEST_TASK_ID, TEST_TASK_ID_ALTERNATE]))}'"
                f" in DAG '{TEST_DAG_ID}' failed.",
            ),
            (
                {"failed_states": [State.FAILED]},
                f"The external DAG {TEST_DAG_ID} failed.",
            ),
        ),
    )
```

Reply:
@Lee-W I have added your changes in PR. Please take a look